vllm/model_executor/models/llama.py

```diff
@@ -545,6 +553,8 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
                 weight_loader = getattr(param, "weight_loader",
                                         default_weight_loader)
                 weight_loader(param, loaded_weight)
+            if current_platform.is_hpu():
+                torch.hpu.synchronize()
```

Comment:
Can we remove this hack by hooking into the weight loader? For example, we do a similar thing for TPUs: https://github.com/vllm-project/vllm/blob/7f60520deb05d2e097b408e3310f1d383fbf1de6/vllm/model_executor/model_loader/loader.py#L358

Reply:
I've removed this workaround completely for now - it was needed only for Llama-405B